heft-plugins/heft-localization-typings-plugin/src/LocalizationTypingsPlugin.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type {
+  HeftConfiguration,
+  IHeftTaskPlugin,
+  IHeftTaskRunIncrementalHookOptions,
+  IHeftTaskSession,
+  IScopedLogger,
+  IWatchedFileState
+} from '@rushstack/heft';
+import { TypingsGenerator } from '@rushstack/localization-utilities';
+import { Path } from '@rushstack/node-core-library';
+
+export interface ILocalizationTypingsPluginOptions {
+  /**
+   * Setting this option wraps the typings export in a default property.
+   */
+  exportAsDefault?: boolean;
+
+  /**
+   * Additional folders, relative to the project root, where the generated typings should be emitted to.
+   */
+  secondaryGeneratedTsFolders?: string[];
+
+  /**
+   * When `exportAsDefault` is true and this option is true, the default export interface name will be inferred
+   * from the filename.
+   */
+  inferDefaultExportInterfaceNameFromFilename?: boolean;
+
+  /**
+   * When `exportAsDefault` is true, this value is placed in a documentation comment for the
+   * exported default interface. Ignored when `exportAsDefault` is false.
+   */
+  exportAsDefaultDocumentationComment?: string;
+
+  /**
+   * An array of string names to ignore when generating typings.
+   */
+  stringNamesToIgnore?: string[];
+}
+
+const PLUGIN_NAME: 'localization-typings-plugin' = 'localization-typings-plugin';
+
+export default class LocalizationTypingsPlugin implements IHeftTaskPlugin<ILocalizationTypingsPluginOptions> {
+  public apply(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration,
+    options?: ILocalizationTypingsPluginOptions
+  ): void {
+    const slashNormalizedBuildFolderPath: string = Path.convertToSlashes(heftConfiguration.buildFolderPath);
```

Comment:
Should we just do this normalization inside of Heft? I think we're doing this in lots of separate plugins at this point.